### PR TITLE
Iteration 3 henley

### PR DESCRIPTION
**Pull request by HenleyCH**
Refactored. Executes. Ready for mediation.

### CODE REVIEW

File: Library.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+    public member addMember(String lastName, String firstName, String email, int phoneNo) {
+        int nextMemberId = this.getNextMemberId();
+        member newMember = new member(lastName, firstName, email, phoneNo, nextMemberId);
+        members.put(newMember.GeT_ID(), newMember);
+        return newMember;
+    }
+
+
+    public Book addBook(String a, String t, String c) {
```
Comment (cmougios-ITC205) [inline Library.java]: All variable names are to be meaningful

Comment (cmougios-ITC205) [inline Library.java]: D1:All variable names are to be meaningful

File: Library.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        int nextBookId = this.getNextBookId();
+        Book newBook = new Book(a, t, c, nextBookId);
```
Comment (cmougios-ITC205) [inline Library.java]: D2:All variable names are to be meaningful

File: Library.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    public int getLoanLimit() {
+        return LOAN_LIMIT;
+    }
+
+
+    public boolean canMemberBorrow(member currentMember) {
+        if (currentMember.Number_Of_Current_Loans() == LOAN_LIMIT)
+            return false;
+
+        if (currentMember.Fines_OwEd() >= MAX_FINES_OWED)
```
Comment (cmougios-ITC205) [inline Library.java:171]: D3:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+            return false;
+
+        for (loan loanBeingAssessed : currentMember.GeT_LoAnS())
+            if (loanBeingAssessed.OVer_Due())
```
Comment (cmougios-ITC205) [inline Library.java:175]: D4:All method names are to start with a lowercase letter and to be in camelBack. Name should also contain verb phrases e.g. isOverDue()

File: Library.java [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+
+    public boolean canMemberBorrow(member currentMember) {
+        if (currentMember.Number_Of_Current_Loans() == LOAN_LIMIT)
+            return false;
+
+        if (currentMember.Fines_OwEd() >= MAX_FINES_OWED)
+            return false;
+
+        for (loan loanBeingAssessed : currentMember.GeT_LoAnS())
```
Comment (cmougios-ITC205) [inline Library.java:174]: D5:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    public loan getLoanByBookId(int bookId) {
+        if (currentLoans.containsKey(bookId)) {
+            return currentLoans.get(bookId);
+        }
+        return null;
+    }
+
+
+    public double calculateOverDueFine(loan currentLoan) {
+        if (currentLoan.OVer_Due()) {
```
Comment (cmougios-ITC205) [inline Library.java:210]: D6:All method names are to start with a lowercase letter and to be in camelBack. Name should also contain verb phrases e.g. isOverDue()

File: Library.java [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+            Date loanDueDate = currentLoan.Get_Due_Date();
```
Comment (cmougios-ITC205) [inline Library.java:211]: D7:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+        return 0.0;
+    }
+
+
+    public void dischargeLoan(loan currentLoan, boolean isDamaged) {
+        member loanee = currentLoan.Member();
+        Book loanedBook  = currentLoan.Book();
+
+        double overDueFine = calculateOverDueFine(currentLoan);
+        loanee.Add_Fine(overDueFine);
```
Comment (cmougios-ITC205) [inline Library.java:225]: D8:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+        loanee.dIsChArGeLoAn(currentLoan);
```
Comment (cmougios-ITC205) [inline Library.java:227]: D9:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+        loanedBook.returnBook(isDamaged);
+
+        int loanedBookId = loanedBook.getBookId();
+
+        if (isDamaged) {
+            loanee.Add_Fine(DAMAGE_FEE);
```
Comment (cmougios-ITC205) [inline Library.java:233]: D10:All method names are to start with a lowercase letter and to be in camelBack

File: Library.java [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+            damagedBooks.put(loanedBookId, loanedBook);
+        }
+        currentLoan.DiScHaRgE();
```
Comment (cmougios-ITC205) [inline Library.java:236]: D11:All method names are to start with a lowercase letter and to be in camelBack

**Review by cmougios-ITC205 — COMMENTED** [review]
Please review feedback. I may be incorrect in my comments but please correct me! I also noticed that you used the collection interface. Keen to understand why use this interface instead of the List Interface?

File: Library.java (hunk 3 above)
Comment (HenleyCH) [reply Library.java:171]: Methods 'Number_Of_Current_Loans()' and 'Fines_OwEd()' belong to member.java, not Library.java.

File: Library.java (hunk 4 above)
Comment (HenleyCH) [reply Library.java:175]: Method 'GeT_LoAnS()' belongs to member.java, not Library.java.
Method 'OVer_Due()' belongs to loan.java, not Library.java.

File: Library.java (hunk 5 above)
Comment (HenleyCH) [reply Library.java:174]: Methods 'Fines_OwEd()' and 'GeT_LoAnS()' belong to member.java, not Library.java.

File: Library.java (hunk 6 above)
Comment (HenleyCH) [reply Library.java:210]: Method 'OVer_Due()' belongs to loan.java, not Library.java.

File: Library.java (hunk 7 above)
Comment (HenleyCH) [reply Library.java:211]: Method 'Get_Due_Date()' belongs to loan.java, not Library.java.

File: Library.java (hunk 8 above)
Comment (HenleyCH) [reply Library.java:225]: Method 'Book()' belongs to loan.java, not Library.java.
Method 'Add_Fine()' belongs to member.java, not Library.java.

File: Library.java (hunk 9 above)
Comment (HenleyCH) [reply Library.java:227]: Method 'Add_Fine()' belongs to member.java, not Library.java.
Method 'dIsChArGeLoAn()' belongs to member.java, not Library.java.

File: Library.java (hunk 10 above)
Comment (HenleyCH) [reply Library.java:233]: Method 'Add_Fine()' belongs to member.java, not Library.java.

File: Library.java (hunk 11 above)
Comment (HenleyCH) [reply Library.java:236]: Method 'DiScHaRgE()' belongs to loan.java, not Library.java.

**Review by cmougios-ITC205 — APPROVED** [review]

**Pull request merged**